seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/CannalToKafakIT.java

```diff
@@ -0,0 +1,146 @@
+        MySqlContainer mySqlContainer = new MySqlContainer(version)
+            .withConfigurationOverride("docker/server-gtids/my.cnf")
+            .withSetupSQL("docker/setup.sql")
+            .withNetwork(NETWORK)
+            .withNetworkAliases(MYSQL_HOST)
+            .withDatabaseName("canal")
+            .withUsername("st_user")
+            .withPassword("seatunnel")
+            .withLogConsumer(new Slf4jLogConsumer(LOG));
+        mySqlContainer.setPortBindings(com.google.common.collect.Lists.newArrayList(
+            String.format("%s:%s", MYSQL_PORT, MYSQL_PORT)));
+        return mySqlContainer;
+    }
+
+    private void createCanalContainer() {
+        CANAL_CONTAINER = new GenericContainer<>(CANAL_DOCKER_IMAGE)
+            .withCopyFileToContainer(MountableFile.forClasspathResource("canal/canal.properties"), "/app/server/conf/canal.properties")
+            .withCopyFileToContainer(MountableFile.forClasspathResource("canal/instance.properties"), "/app/server/conf/example/instance.properties")
+            .withNetwork(NETWORK)
+            .withNetworkAliases(CANAL_HOST)
+            .withCommand()
+            .withLogConsumer(new Slf4jLogConsumer(DockerLoggerFactory.getLogger(CANAL_DOCKER_IMAGE)));
+        CANAL_CONTAINER.setPortBindings(com.google.common.collect.Lists.newArrayList(
+            String.format("%s:%s", CANAL_PORT, CANAL_PORT)));
+    }
+
+    private void createKafkaContainer(){
+        KAFKA_CONTAINER = new KafkaContainer(DockerImageName.parse(KAFKA_IMAGE_NAME))
+            .withNetwork(NETWORK)
+            .withNetworkAliases(KAFKA_HOST)
+            .withLogConsumer(new Slf4jLogConsumer(DockerLoggerFactory.getLogger(KAFKA_IMAGE_NAME)));
+        KAFKA_CONTAINER.setPortBindings(com.google.common.collect.Lists.newArrayList(
+            String.format("%s:%s", KAFKA_PORT, KAFKA_PORT)));
+    }
+
+    @BeforeAll
+    @Override
+    public void startUp() {
+
+        LOG.info("The third stage: Starting Kafka containers...");
+        createKafkaContainer();
+        Startables.deepStart(Stream.of(KAFKA_CONTAINER)).join();
+        LOG.info("Containers are started");
+
+        LOG.info("The first stage: Starting Mysql containers...");
+        Startables.deepStart(Stream.of(MYSQL_CONTAINER)).join();
+        LOG.info("Containers are started");
+
+        LOG.info("The first stage: Starting Canal containers...");
+        createCanalContainer();
+        Startables.deepStart(Stream.of(CANAL_CONTAINER)).join();
+        LOG.info("Containers are started");
+
+        inventoryDatabase.createAndInitialize();
+    }
+
+    @TestTemplate
+    public void testCannalToKafakCannalFormatAnalysis(TestContainer container) throws IOException, InterruptedException {
+        Container.ExecResult execResult = container.executeJob("/kafkasource_canal_to_console.conf");
+        Assertions.assertEquals(0, execResult.getExitCode(), execResult.getStderr());
```

Comment:
How to validate data output by sink

Reply:
reference
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcSinkCDCChangelogIT.java#L87